library/src/scripts/forms/themeEditor/ColorPicker.tsx

```diff
@@ -0,0 +1,154 @@
+/*
+ * @author Stéphane LaFlèche <stephane.l@vanillaforums.com>
+ * @copyright 2009-2019 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+import React, { useMemo, useRef, useState } from "react";
+import { visibility } from "@library/styles/styleHelpersVisibility";
+import classNames from "classnames";
+import { colorPickerClasses } from "@library/forms/themeEditor/colorPickerStyles";
+import { color, ColorHelper } from "csx";
+import { colorOut } from "@library/styles/styleHelpersColors";
+import { useField } from "formik";
+import Button from "@library/forms/Button";
+import { ButtonTypes } from "@library/forms/buttonStyles";
+import { t } from "@vanilla/i18n/src";
+import { uniqueIDFromPrefix } from "@library/utility/idUtils";
+import { themeBuilderClasses } from "@library/forms/themeEditor/themeBuilderStyles";
+
+const Collit = require("collit");
+
+type IErrorWithDefault = string | boolean; // Uses default message if true
+
+export interface IColorPicker {
+    inputProps?: React.HTMLAttributes<Omit<Omit<Omit<HTMLInputElement, "type">, "id">, "tabIndex">>;
+    inputID: string;
+    variableID: string;
+    labelID: string;
+    defaultValue?: ColorHelper;
+    inputClass?: string;
+    errors?: IErrorWithDefault[]; // Uses default message if true
+}
+
+export const getDefaultOrCustomErrorMessage = (error: string | true, defaultMessage: string) => {
+    return typeof error === "string" ? error : defaultMessage;
+};
+
+export default function ColorPicker(props: IColorPicker) {
+    const classes = colorPickerClasses();
+    const colorInput = useRef<HTMLInputElement>(null);
+    const textInput = useRef<HTMLInputElement>(null);
+    const builderClasses = themeBuilderClasses();
+
+    const errorID = useMemo(() => {
+        return uniqueIDFromPrefix("colorPickerError");
+    }, []);
+
+    const initialColor = props.defaultValue ? props.defaultValue : color("#fff");
+
+    const [lastValidColor, setLastValidColor] = useState(initialColor); // Always "Color" object
+    const [inputTextValue, setInputTextValue] = useState(initialColor.toString()); // Always string
```

Comment:
This value should be stored on `selectedColor.value` from formik already. I'm not sure we need to store it again.